New article: why you should be using new css features today

brechtDR opened the pull request:
This is a new article on why you should be using new css features today. Ready for review.

This is part 1/2

brechtDR committed:
New article: why you should be using new css features today

brechtDR committed:
fix date typo

brechtDR committed:
fix title typo

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+## The main benefits of using the latest CSS features
+
+Is it because they are cool, new and shiny? No, there is a lot more going on than that. Some of the latest things happening in CSS solve some of the biggest frustrations in web development. Just imagine some of the recent features that are starting to see the light of day, such as container queries, :has(), cascade layers, logical properties, masonry and scroll animations. Although some are more  ...
```
Perhaps links to articles about these subjects already available on the TechHub?

Shameless plug: https://techhub.iodigital.com/articles/container-queries-the-next-step-towards-a-truly-modular-css

mrtnvh replied:
javascript -> JavaScript

brechtDR replied:
> https://techhub.iodigital.com/articles/container-queries-the-next-step-towards-a-truly-modular-css

I don't mind adding the link there. But i'm afraid that the article has become a bit outdated since it's still having container declarations without a specified name attached to it.

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+This is a personal opinion but in my experience, I noticed that when you just read about a new feature, you start to forget about it when it gets released and you won’t be picking it up. Thus, you might be writing outdated code and actually doing more work than you should. Being lazy can be a good trait for a developer, so write less code that does more. If you’ve experimented with it, chances ar ...
+
+But I hear you, you will be doing some more work while learning these new features. You will have to write some support flags in your CSS, you might need to set up some extra packages in your postCSS config, but believe me, your future self will be grateful!
```
postCSS -> PostCSS

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
````diff
@@ -0,0 +1,178 @@
+}
+```
+
+If you are paying attention you might notice that my code just got larger and you’re right. Even the output of my code will be larger by a small margin. But I just received a newsflash from the client:
````
Add a comma after `attention`

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+**Hey, we were thinking about adding a dark theme for our website, the alerts should have a light variant background in that case. Also, we would like them to be black and white for people who prefer high contrast in the browser.**
+
+If you were using sass variables, you’d probably have to re-declare all your alerts scoped into media queries. But, because I used custom properties I can re-declare my custom properties:
```
sass -> Sass

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+My output CSS is now significantly smaller as I didn't have to redo all the alert selectors based on the dark theme. It also keeps my CSS modular and flexible for future updates. My future self is now a happy camper. I know making all your alerts black and white isn’t good A11Y practice, it was just to prove my point when it comes to custom properties with this very simple example.
+
+Also, CSS custom properties can be set by using JS, it makes playing around with css so much easier and doesn’t require you to set a whole bunch of inline styles on your dom. Just [take a look at this very simple example on codepen](https://codepen.io/utilitybend/pen/JjvoYGj).
```
`makes playing around with css so much easier` -> `makes playing around with CSS so much easier`

mrtnvh replied:
codepen -> CodePen

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+### Where we are today: Designers should just “respect the grid”
+
+A lot of people (including myself) have been using a 12 column Bootstrap flexbox grid for so long that it has become a second nature. Although design consistency is important when it comes to time management of front-end development and creating a good design system, a 12 column grid seems to be a bit of an outdated idea. Especially now that CSS-grid has such great support with “subgrid” just aro ...
```
12 column -> 12-column
a second nature -> second nature

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
````diff
@@ -0,0 +1,178 @@
+}
+```
+
+You might wonder why we should make things harder than it used to be by declaring multiple grids on a website. \
````
than it used to be -> than they used to be

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
````diff
@@ -0,0 +1,178 @@
+```
+
+You might wonder why we should make things harder than it used to be by declaring multiple grids on a website. \
+With some careful CSS planning in your design system you can easily define your common patterns, so in the end, it’s only a small effort to create a front-end design system that’s a bit more unique and “designy”. You can also easily scope these grids based on your component to make it bulletproof for other tweaks on the design system. Especially combined with custom properties you can make someth ...
````
nicely contained to the component -> nicely contained in the component

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+### Where we are today: not a CSS purist yet
+
+There is still a lot missing at the moment before we can truly step away from SASS, especially when it comes to using variables in things such as media queries. But that doesn’t mean we can’t combine these two technologies. I love the idea behind Bootstrap and Tailwind as well, both of them are fantastic tools to get something done quickly, but as someone who likes a challenge when it comes to de ...
```
SASS -> Sass

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+
+There is still a lot missing at the moment before we can truly step away from SASS, especially when it comes to using variables in things such as media queries. But that doesn’t mean we can’t combine these two technologies. I love the idea behind Bootstrap and Tailwind as well, both of them are fantastic tools to get something done quickly, but as someone who likes a challenge when it comes to de ...
+
+I’m also a strong believer that clients will ask for more unique designs in the future, because at the moment there are a lot of webshops out there and a lot of them look more and less the same. Design is emotion and emotion sells, and if every emotion looks the same, it’s a pale world (wide web).
```
Don't need a comma between `future` and `because`

mrtnvh commented on data/blog/why-you-should-be-using-new-css-features-today/why-you-should-be-using-new-css-features-today.md:
```diff
@@ -0,0 +1,178 @@
+- Maybe I should comment on Slack that Brecht is a diluted idiot who lives in a fantasy where time and money doesn’t matter?
+- “CSS IS AWESOME!” (no, _YOU_ are awesome!)
+
+In no case, am I assaulting any use of frameworks, libraries and maybe tools that can purge your CSS from unneeded selectors. It’s about reflecting on how we can take our CSS game to the next level by evolving as a developer and maybe even become a fully fledged CSS-developer.
```
become a fully fledged CSS-developer -> becoming a fully-fledged CSS-developer

mrtnvh reviewed (changes requested):
Thanks for this article @brechtDR, looking great! I spotted some inconsistencies with capitalization on brand names and some typos. Can you fix these?

brechtDR committed:
add secondary blogpost in draft

brechtDR committed:
fix inconsitencies

brechtDR commented:
Hey @vanhoofmaarten 

Thanks for the review.
I also added part 2 (was still checking this with Sander to see if that was possible "in draft")

I fixed most of the issues. The container queries blogpost seems to be a bit outdated at the moment. 
Maybe something you can look in to? I could add some links to it in part 2 then?

DaveBitter commented:
Nice! Could you maybe turn it into a series which makes it easier to navigate in the UI? You can refer to, for instance, this PR https://github.com/iodigital-com/io-technology/pull/111 to see how

brechtDR committed:
add links

brechtDR committed:
turn posts into a serie

brechtDR committed:
create series
